### PR TITLE
doc/user: improve SQL client documentation

### LINKED ISSUE #22409: doc/user: Reconcile issues with two "SQL Clients" sections
### Documentation request

I was just prepping @joacoc  post for publish, I noticed we have a bit of a disconnect between:

https://materialize.com/docs/integrations/#sql-clients and
https://materialize.com/docs/integrations/sql-clients/

### Issues:

1. **Overview Page:** "SQL Clients" link in psql line item links to itself, not the actual sql-clients page image
2. **Overview Page** DataGrip and PopSQL have one-liner explanations on the overview page, but more detail on the SQL Clients page, maybe link from overview to SQL clients?
1. **SQL Clients Page** - Doesn't provide as complete a list as the overview, PopSQL, TablePlus are missing. Someone navigating directly to this page might incorrectly conclude that these are not supported.

### Affected Pages

https://materialize.com/docs/integrations/#sql-clients and
https://materialize.com/docs/integrations/sql-clients/

### Related work

https://github.com/MaterializeInc/materialize/issues/19891

### PR DESCRIPTION
**Pull request by morsapaes**
Fixes #22409, #18235.

Once the integrations page is revamped (#23398), we'll yank the tool specific information out of this page and into tool-specific guides, and visually list all supported SQL clients here with a link.

### CODE REVIEW

File: doc/user/content/integrations/sql-clients.md [hunk 1]
```diff
@@ -17,7 +17,136 @@ with many SQL clients that support PostgreSQL (see [Tools and Integrations](/int
...
+Before connecting, double-check that you've created an **app-password** for your
+user. This password is auto-generated, and prefixed with `mzp_`.
+
+### Runtime connection parameters
+
+{{< warning >}}
+Parameters set in the connection string work for the **lifetime of the
+session**, but do not affect other sessions. To permanently change the default
+value of a session variable for a specific user (i.e. role), use the
+[`ALTER ROLE...SET`](/sql/alter-role) command.
```
Comment (chaas) [inline doc/user/content/integrations/sql-clients.md:44]: Heads up that this isn't in public preview quite yet, the last changes just merged so will go out with next week's release.

**Review by chaas — APPROVED** [review]
Should we also mention the VS Code extension here?

File: doc/user/content/integrations/sql-clients.md (hunk 1 above)
Comment (morsapaes) [reply doc/user/content/integrations/sql-clients.md:44]: It is now! 🎉

**Comment by morsapaes** [comment]
Thanks, @chaas! I'm hoping to remove any tool-specific instructions from this page with #23398, but I've at least added VSCode to the list of integrations for now + opened #24892 since we don't yet have a standalone guide for it.

File: doc/user/content/integrations/sql-clients.md [hunk 2]
```diff
@@ -17,7 +17,136 @@ with many SQL clients that support PostgreSQL (see [Tools and Integrations](/int
...
+{{< /warning >}}
+
+You can pass runtime connection parameters (like `cluster`, `isolation_level`,
```
Comment (chuck-alt-delete) [inline doc/user/content/integrations/sql-clients.md:47]: I just want to note that users might not find these session variable options if they are looking in the sql drivers section to connect programmatically. Maybe we could link here in the sql driver pages so they can see how to include the more advanced options in the connection string

Comment (morsapaes) [reply doc/user/content/integrations/sql-clients.md:47]: Will do that as a follow-up!

**Pull request merged**